src/LinuxTracing/UprobesUnwindingVisitor.cpp

```diff
@@ -129,9 +130,12 @@ void UprobesUnwindingVisitor::visit(CallchainSamplePerfEvent* event) {
 
   Callstack* callstack = sample.mutable_callstack();
 
-  // The top of a callchain is always inside the kernel code and we don't expect samples to be only
-  // inside the kernel.
-  if (event->GetCallchainSize() <= 1) {
+  // Callstacks with only two frames (the first is in the kernel, the second is the sampled address)
+  // are unwinding errors.
+  // Note that this doesn't exclude samples inside the main function of any thread as the main
+  // function is never the outermost frame. For example, for the main thread the outermost function
+  // is _start, followed by __libc_start_main. For other threads, the outermost function is clone.
+  if (event->GetCallchainSize() <= 2) {
```

Comment:
Just a note that I'm planning to split this in two, `== 2` as unwinding error and `== 1` as unexpected (there should always be at least one (real) frame).

Reply:
Sorry 😅